dts/bindings/rtc/nxp,irtc.yaml

```diff
@@ -40,3 +39,9 @@ properties:
       3 <- Coarse 1Hz
       Default value ensures the output clock is turned off.
       This is the reset value.
+  
+  share-counter:
+    type: boolean
+    description: |
+      This slave irtc instance shares the data and time counters of the master irtc instance. 
+      This means the code cannot set the data and time counters, but can only read them.
```

Comment:
This property should be enforced on the driver level, by implementing the set_time API like
```
static int set_time_not_perm(const struct device *dev, const struct rtc_time *timeptr)
{
        ARG_UNUSED(timeptr);
        return -EPERM;
}

static DEVICE_API(...) = {
        .set_time = set_time_not_sup,
        ...
};
```
the sample does not need to be updated, it will correclty print "Cannot write date time: -1"